### ISSUE TITLE
Could not find *gem* in any of the sources w/t require "bundler/setup" 

### ISSUE THREAD

**Issue by Mongey** [navarasu/serverless-ruby-layer#36]
**Describe the bug**

To auto require gems, we specific the following.
```ruby
  require "rubygems"
  require "bundler/setup"
```
I can remove these requires, and it works, but, wondering why it doesn't

**Error Stack Trace**

```
Init error when loading handler handler.Handler.list
2021-02-22T14:55:03.403+00:00	{
2021-02-22T14:55:03.403+00:00	"errorMessage": "Could not find public_suffix-3.1.1 in any of the sources",
2021-02-22T14:55:03.403+00:00	"errorType": "Init<Bundler::GemNotFound>",
2021-02-22T14:55:03.403+00:00	"stackTrace": [
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/spec_set.rb:87:in `block in materialize'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/spec_set.rb:81:in `map!'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/spec_set.rb:81:in `materialize'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/definition.rb:175:in `specs'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/definition.rb:245:in `specs_for'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/definition.rb:227:in `requested_specs'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/runtime.rb:91:in `block in definition_method'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/runtime.rb:20:in `setup'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler.rb:148:in `setup'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/setup.rb:20:in `block in <top (required)>'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/ui/shell.rb:136:in `with_level'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/ui/shell.rb:88:in `silence'",
2021-02-22T14:55:03.403+00:00	"/var/runtime/gems/bundler-2.2.5/lib/bundler/setup.rb:20:in `<top (required)>'",
2021-02-22T14:55:03.403+00:00	"/var/lang/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
2021-02-22T14:55:03.403+00:00	"/var/lang/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
2021-02-22T14:55:03.403+00:00	"/var/task/config/initializers/autoloader.rb:11:in `<top (required)>'",
2021-02-22T14:55:03.403+00:00	"/var/lang/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
2021-02-22T14:55:03.403+00:00	"/var/lang/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
2021-02-22T14:55:03.403+00:00	"/var/task/handler.rb:1:in `<top (required)>'",
```

**Comment by navarasu** [comment]
Can you share the `serverless.yml` (Need ruby_layer config to check whether you are using local bundler or docker version)?
If local give your local bundler version and ruby version.

**Comment by Mongey** [comment]
@navarasu 

```yaml
plugins:
  - serverless-ruby-layer

package:
  include:
    - Gemfile
    - Gemfile.lock
    - handler.rb
    - config/**
    - app/**
    - ./

custom:
  rubyLayer:
    use_docker: true
```

```

RUBY VERSION
   ruby 2.5.8p224

BUNDLED WITH
   2.1.4
```

**Comment by navarasu** [comment]
Thanks. Is your aws runtime version 2.5 or 2.7?
`
provider​:
  ​name​: ​aws​
  ​runtime​: ​ruby2.5​
`

**Comment by Mongey** [comment]
Whoops, I forgot that part, It's `2.5`

**Comment by jagthedrummer** [comment]
I think this is the same problem I'm seeing in #40, and it should hopefully be fixed by #41. (I haven't quite figured out how to run a local copy of a serverless plugin, so I haven't tested it directly. 😬 )

**Comment by navarasu** [comment]
@Mongey I dont think  issue #40 is related to this ticket.

**Comment by navarasu** [comment]
@Mongey I am not sure that issue #40 is related to your issue. Currently created an example for your case and trying to reproduce the issue in my local with 1.4.0 version

**Comment by navarasu** [comment]
With existing serverless version (1.4.0) itself, it tried the requires all example. It is working.

Below is the example I have tried. The folder contains the following four files,
```sh
➜  basic-requires-all tree
.
├── Gemfile
├── Gemfile.lock
├── handler.rb
└── serverless.yml

0 directories, 4 files
```
**Gemfile**
```ruby
source 'https://rubygems.org'
gem 'httparty'
```
**Gemfile.lock**
```ruby
GEM
  remote: https://rubygems.org/
  specs:
    httparty (0.18.1)
      mime-types (~> 3.0)
      multi_xml (>= 0.5.2)
    mime-types (3.3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2021.0225)
    multi_xml (0.6.0)

PLATFORMS
  ruby

DEPENDENCIES
  httparty

BUNDLED WITH
   2.1.4
```
**handler.rb**

```ruby
require 'rubygems'
require 'bundler/setup'
Bundler.require(:default)

def hello(event:, context:)
  body = HTTParty.get("https://github.com").body

  { statusCode: 200, body: body }
end

```

**serverless.yml**

```yml
service: basic 

plugins:
  - serverless-ruby-layer
provider:
  name: aws
  runtime: ruby2.5

custom:
  rubyLayer:
    use_docker: true

package:
  include:
    - Gemfile
    - Gemfile.lock

functions:
  hello:
    handler: handler.hello
```
 I am not able to reproduce your issue. So please share with me an example to reproduce your issue.

**Comment by navarasu** [comment]
@Mongey  I am not sure that issue #40 is related to your issue. Currently, I have created an example for your case and tried to reproduce the issue in my local with the 1.4.0 version.

With the existing serverless version (1.4.0) itself, the `requires all` example. It is working.
Below is the example I have tried. The folder contains the following four files,
```sh
➜  basic-requires-all tree
.
├── Gemfile
├── Gemfile.lock
├── handler.rb
└── serverless.yml

0 directories, 4 files
```
**Gemfile**
```ruby
source 'https://rubygems.org'
gem 'httparty'
```
**Gemfile.lock**
```ruby
GEM
  remote: https://rubygems.org/
  specs:
    httparty (0.18.1)
      mime-types (~> 3.0)
      multi_xml (>= 0.5.2)
    mime-types (3.3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2021.0225)
    multi_xml (0.6.0)

PLATFORMS
  ruby

DEPENDENCIES
  httparty

BUNDLED WITH
   2.1.4
```
**handler.rb**

```ruby
require 'rubygems'
require 'bundler/setup'
Bundler.require(:default)

def hello(event:, context:)
  body = HTTParty.get("https://github.com").body

  { statusCode: 200, body: body }
end

```

**serverless.yml**

```yml
service: basic 

plugins:
  - serverless-ruby-layer
provider:
  name: aws
  runtime: ruby2.5

custom:
  rubyLayer:
    use_docker: true

package:
  include:
    - Gemfile
    - Gemfile.lock

functions:
  hello:
    handler: handler.hello
```
 I am not able to reproduce your issue. So please share with me an example to reproduce your issue.

**Comment by navarasu** [comment]
Hi @Mongey, I am able to reproduce the issue by replacing it with a different version for a dependency gem in  `Gemfile.lock`.
. In the above example, I have replaced it with the following `Gemfile.lock`

**Gemfile.lock**
```
GEM
  remote: https://rubygems.org/
  specs:
    httparty (0.18.1)
      mime-types (~> 3.0)
      multi_xml (>= 0.5.2)
    mime-types (3.3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2020.1104)
    mini_portile2 (2.5.1)
    multi_xml (0.6.0)
    nokogiri (1.11.3)
      mini_portile2 (~> 2.5.0)
      racc (~> 1.4)
    racc (1.5.2)

PLATFORMS
  ruby

DEPENDENCIES
  httparty
  nokogiri

BUNDLED WITH
   2.1.4
```
In Gemfile.lock, the mime-types-data gem version is `3.2020.1104`. But the packed gem contains `3.2021.0225`. I deployed the change. Now I get the below error,

```
{
    "errorMessage": "Could not find mime-types-data-3.2020.1104 in any of the sources",
    "errorType": "Init<Bundler::GemNotFound>",
    "stackTrace": [
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/spec_set.rb:89:in `block in materialize'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/spec_set.rb:82:in `map!'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/spec_set.rb:82:in `materialize'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/definition.rb:185:in `specs'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/definition.rb:255:in `specs_for'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/definition.rb:237:in `requested_specs'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/runtime.rb:91:in `block in definition_method'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/runtime.rb:20:in `setup'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler.rb:148:in `setup'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/setup.rb:20:in `block in <top (required)>'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/ui/shell.rb:136:in `with_level'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/ui/shell.rb:88:in `silence'",
        "/var/runtime/gems/bundler-2.2.15/lib/bundler/setup.rb:20:in `<top (required)>'",
        "/var/lang/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
        "/var/lang/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
        "/var/task/handler.rb:2:in `<top (required)>'",
        "/var/lang/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:85:in `require'",
        "/var/lang/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:85:in `require'"
    ]
}
```

**Comment by navarasu** [comment]
* Added example and test for `bundler.require` all.
* Include `Gemfile` and `Gemfile.lock` in the function package if the lock file is present. 

So the below config **not required**
```yml
package:
  include:
    - Gemfile
    - Gemfile.lock
 ```
It will auto-include `Gemfile` and `Gemfile.lock` if  `Gemfile.lock` is present.  Refer example  [here](https://github.com/navarasu/serverless-ruby-layer/blob/master/examples/bundler-require-all) 
* Added test case for Gemfile and Gemfile.lock auto include

**Comment by navarasu** [comment]
Fix #40 solves this ticket. I will update the doc and will create a new release

**Comment by navarasu** [comment]
@Mongey  This fix is included in release 1.5.0. Please update the npm version and verify.
```
npm i serverless-ruby-layer
```
